BookingQueueSubscriber/BookingQueueSubscriber.Services/MessageHandlers/EndpointUpdatedHandler.cs

```diff
@@ -11,26 +13,57 @@ namespace BookingQueueSubscriber.Services.MessageHandlers
     public class EndpointUpdatedHandler : IMessageHandler<EndpointUpdatedIntegrationEvent>
     {
         private readonly IVideoApiService _videoApiService;
+        private readonly ILogger<EndpointUpdatedHandler> _logger;
+        private const int RetryLimit = 3;
+        private const int RetrySleep = 3000;
 
-        public EndpointUpdatedHandler(IVideoApiService videoApiService)
+        public EndpointUpdatedHandler(IVideoApiService videoApiService, ILogger<EndpointUpdatedHandler> logger)
         {
             _videoApiService = videoApiService;
+            _logger = logger;
         }
 
         public async Task HandleAsync(EndpointUpdatedIntegrationEvent eventMessage)
         {
-            var conference = await _videoApiService.GetConferenceByHearingRefId(eventMessage.HearingId);
             ParticipantDetailsResponse defenceAdvocate = null;
+            ConferenceDetailsResponse conference = null;
+
             if (!string.IsNullOrEmpty(eventMessage.DefenceAdvocate))
             {
-                defenceAdvocate = conference.Participants.Single(x => x.ContactEmail ==
-                    eventMessage.DefenceAdvocate);
+                for (var retry = 0; retry <= RetryLimit; retry++)
+                {
+                    conference = await _videoApiService.GetConferenceByHearingRefId(eventMessage.HearingId);
+
+                    defenceAdvocate = conference.Participants.SingleOrDefault(x => x.ContactEmail ==
```

Comment:
I think we should check conference for null

Reply:
Added if state on conference to log error should it return a null value. Otherwise it will progress through code as normal